Automattic/jetpack
# Launchpad Checklist API Refactor/Fixes: Task completion status and performance optimizations

Fixes https://github.com/Automattic/wp-calypso/issues/76058

## Proposed changes:

* This PR adds changes to sync WPCOM Launchpad code with Jetpack
* Use with https://github.com/Automattic/wp-calypso/pull/76467 to use launchpad checklist API on Calypso
* Task attributes ('completion', 'disabled', etc.) are now determined when the user makes a request to the Launchpad checklist API.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Checkout Calypso PR with updated launchpad checklist API changes (https://github.com/Automattic/wp-calypso/pull/76467)
* Sandbox public api and your test site(s).
* Run `bin/jetpack-downloader test jetpack-mu-wpcom-plugin add/launchpad-completion-logic` on your sandbox.
* Test the different launchpad flows (newsletter, link-in-bio, write, build, free, videopress) and ensure that the launchpad checklist tasks are marked as completed correctly when you go through them. Keep the production version of the launchpad open in a different tab and compare it with calypso localhost launchpad as you walk through each checklist task.
